tfx/examples/iris/iris_resolver_e2e_test.py

```diff
@@ -0,0 +1,214 @@
+from tfx.components import StatisticsGen
+from tfx.components import Trainer
+from tfx.components.base import executor_spec
+from tfx.components.trainer.executor import GenericExecutor
+from tfx.dsl.experimental import latest_artifacts_resolver
+from tfx.orchestration import metadata
+from tfx.orchestration import pipeline
+from tfx.orchestration.beam.beam_dag_runner import BeamDagRunner
+from tfx.types.standard_artifacts import Schema
+from tfx.types.standard_artifacts import Examples
+from tfx.types.channel import Channel
+from tfx.proto import example_gen_pb2
+from tfx.proto import trainer_pb2
+
+from tfx.utils import path_utils
+from tfx.utils import io_utils
+
+
+def _create_example_pipeline(pipeline_name: Text, pipeline_root: Text,
+                             data_root: Text, metadata_path: Text,
+                             beam_pipeline_args: List[Text]
+                             ) -> pipeline.Pipeline:
+  """Simple pipeline to ingest data into Examples artifacts."""
+  # Brings data into the pipeline or otherwise joins/converts training data.
+  input_config = example_gen_pb2.Input(splits=[
+      example_gen_pb2.Input.Split(name='single_split',
+                                  pattern='span{SPAN}/*')])
+  example_gen = CsvExampleGen(input_base=data_root,
+                              input_config=input_config)
+
+  # Computes statistics over data for visualization and example validation.
+  statistics_gen = StatisticsGen(
+      examples=example_gen.outputs['examples'])
+
+  # Generates schema based on statistics files.
+  schema_gen = SchemaGen(
+      statistics=statistics_gen.outputs['statistics'], infer_feature_shape=True)
+
+  return pipeline.Pipeline(
+      pipeline_name=pipeline_name,
+      pipeline_root=pipeline_root,
+      components=[
+          example_gen,
+          statistics_gen,
+          schema_gen
+      ],
+      enable_cache=False,
+      metadata_connection_config=metadata.sqlite_metadata_connection_config(
+          metadata_path),
+      beam_pipeline_args=beam_pipeline_args)
+
+def _create_trainer_pipeline(pipeline_name: Text, pipeline_root: Text,
+                             module_file: Text, metadata_path: Text,
+                             window_size: int, beam_pipeline_args: List[Text],
+                             ) -> pipeline.Pipeline:
+  """Trainer pipeline to train based on resolver outputs"""
+  # Get latest schema for training.
+  schema_resolver = ResolverNode(
+      instance_name='schema_resolver',
+      resolver_class=latest_artifacts_resolver.LatestArtifactsResolver,
+      schema=Channel(type=Schema))
+
+  # Resolve latest two example artifacts into one channel for trainer.
+  latest_examples_resolver = ResolverNode(
+      instance_name='latest_examples_resolver',
+      resolver_class=latest_artifacts_resolver.LatestArtifactsResolver,
+      resolver_configs={'desired_num_of_artifacts': window_size},
+      latest_n_examples=Channel(type=Examples))
```

Comment:
could you verify the if we put examplegen and resolver in the same pipeline, we can make sure resolver depends on examplegen

window size 3:
pipeline1: examplegen -> tdfv
pipeline2:examplegen ->tfdv
pipeline3: examplegen->tfdv ->resolver->trainer

Reply:
Updated test to also test case for dependency.

Reply:
From running this "full pipeline" and printing out the dependency graph, we get the following:

`I0720 13:11:47.808361 135087534106432 beam_dag_runner.py:151] Component CsvExampleGen depends on [].
I0720 13:11:47.829361 135087534106432 beam_dag_runner.py:165] Component CsvExampleGen is scheduled.
I0720 13:11:47.836360 135087534106432 beam_dag_runner.py:151] Component ResolverNode.latest_examples_resolver depends on ['Run[CsvExampleGen]'].
I0720 13:11:47.836360 135087534106432 beam_dag_runner.py:165] Component ResolverNode.latest_examples_resolver is scheduled.
I0720 13:11:47.843361 135087534106432 beam_dag_runner.py:151] Component StatisticsGen depends on ['Run[CsvExampleGen]'].
I0720 13:11:47.844361 135087534106432 beam_dag_runner.py:165] Component StatisticsGen is scheduled.
I0720 13:11:47.844361 135087534106432 beam_dag_runner.py:151] Component SchemaGen depends on ['Run[StatisticsGen]'].
I0720 13:11:47.851361 135087534106432 beam_dag_runner.py:165] Component SchemaGen is scheduled.
I0720 13:11:47.858361 135087534106432 beam_dag_runner.py:151] Component Trainer depends on ['Run[SchemaGen]', 'Run[ResolverNode.latest_examples_resolver]'].
I0720 13:11:47.858361 135087534106432 beam_dag_runner.py:165] Component Trainer is scheduled.`

Reply:
nice!